### PR TITLE
coqPackages.graph-theory: init at 0.9

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130624` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>coqPackages.graph-theory</li>
  </ul>
</details>

File: pkgs/development/coq-modules/graph-theory/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+mkCoqDerivation {
+  pname = "graph-theory";
+
+  release."0.9".sha256 = "sha256-Hl3JS9YERD8QQziXqZ9DqLHKp63RKI9HxoFYWSkJQZI=";
+
+  releaseRev = v: "v${v}";
+
+  inherit version;
+  defaultVersion = with versions; switch coq.coq-version [
+    { case = isGe "8.12"; out = "0.9"; }
```
Comment (vbgl) [inline pkgs/development/coq-modules/graph-theory/default.nix:15]: Does this build with Coq 8.12?

**Comment by siraben** [comment]
Building when building with Coq 8.12

```
-bash-4.2$ nix log /nix/store/jp6svrgnxwig6xfc3bk8pl9l8jxji0w0-coq8.12-graph-theory-0.9.drv
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/v5i7w30168ih36gkp0zizclv0mpq5rhg-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j32 -l32 SHELL=/nix/store/x0dcb2rxlzf32g0ddfkqqz1sfcyx4yay-bash-4.4-p23/bin/bash
/nix/store/rpdv72w986k9nvj04xk827q9pnhvppjv-coq-8.12.2/bin/coq_makefile -f _CoqProject -o Makefile.coq
make[1]: Entering directory '/tmp/nix-build-coq8.12-graph-theory-0.9.drv-0/source'
COQDEP VFILES
COQC theories/edone.v
COQC theories/bounded.v
COQC theories/setoid_bigop.v
COQC theories/partition.v
COQC theories/preliminaries.v
File "./theories/partition.v", line 28, characters 0-23:
Error: The reference disjointWr was not found in the current environment.

make[2]: *** [Makefile.coq:716: theories/partition.vo] Error 1
make[2]: *** Waiting for unfinished jobs....
File "./theories/preliminaries.v", line 82, characters 0-80:
Error: The reference imset_f was not found in the current environment.

make[2]: *** [Makefile.coq:715: theories/preliminaries.vo] Error 1
make[1]: *** [Makefile.coq:339: all] Error 2
make[1]: Leaving directory '/tmp/nix-build-coq8.12-graph-theory-0.9.drv-0/source'
make: *** [Makefile:2: all] Error 2
```

File: pkgs/development/coq-modules/graph-theory/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/coq-modules/graph-theory/default.nix:15]:
```suggestion
    { case = isGe "8.13"; out = "0.9"; }
```

**Pull request merged**